convert usage properties in network to double (#1624)

nathantypanski opened the pull request:
Fixes #1624.

This is a breaking change, but clients (including the Go client) are already broken without this, as the server is returning double-formatted values.

This leads to issues like [azure-sdk-for-go/issues/906](https://github.com/Azure/azure-sdk-for-go/issues/906).

For all spec versions where the "examples" directory "UsageList.json" item contains double-formatted responses, update the "usage.json" spec to be double-formatted. For earlier versions with integer-formatted responses (2017-06-01), do not change the format/type.

See more discussion from [August 29 last year](https://github.com/Azure/azure-rest-api-specs/pull/1603/files#r135889852) when this issue was discovered.

This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [x] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [ ] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [ ] The spec follows the guidelines described in the [Swagger checklist](../documentation/swagger-checklist.md) document. **(Link breaks for me.)**
  - [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR. **(Validation tools aren't working for me right now. But the change seems simple enough)**

msftclas commented:
[![CLA assistant check](https://cla.opensource.microsoft.com/pull/badge/not_signed)](https://cla.opensource.microsoft.com/Azure/azure-rest-api-specs?pullRequest=2271) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.opensource.microsoft.com/Azure/azure-rest-api-specs?pullRequest=2271) before we can accept your contribution.<br/><br/>:x: nathantypanski [sign now](https://cla.opensource.microsoft.com/Azure/azure-rest-api-specs?pullRequest=2271)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.opensource.microsoft.com/check/Azure/azure-rest-api-specs?pullRequest=2271) it.</sub>

olydis commented:
@ravbhatnagar breaking change

ravbhatnagar commented:
@nathantypanski - needs a new api-version

Commit:
create network api version 2018-01-01

Copied from 2017-11-01.

Commit:
convert usage properties in network to double (#1624)

Fixes #1624.

This is a breaking change, but clients (including the Go client) are
already broken without this, as the server is returning double-formatted
values.

This leads to issues like
[azure-sdk-for-go/issues/906](https://github.com/Azure/azure-sdk-for-go/issues/906).

See more discussion from [August 29 last
year](https://github.com/Azure/azure-rest-api-specs/pull/1603/files#r135889852)
when this issue was discovered.

nathantypanski commented:
Tried making a new network api-version and remade my change in a separate commit. Mind taking another look?

Commit:
add 2018-01 to readme

olydis commented:
@ravbhatnagar ping

ravbhatnagar commented:
Signing off!

olydis commented on specification/network/resource-manager/readme.md:
```diff
@@ -451,6 +481,17 @@ batch:
 
+### Tag: package-2018-01 and python
+
+These settings apply only when `--tag=package-2018-01 --python` is specified on the command line.
```
@lmazuel This look cool? Note `namespace`/`output-folder` have 2017-...

lmazuel replied:
not cool indeed, thanks for being careful!
In this tag, all 2017_11_01 needs to be updated to 2018_01_01

lmazuel replied:
@marstr Please review this PR for Go multiapi

olydis replied:
@marstr @lmazuel before spending too much time here, note [my comment](https://github.com/Azure/azure-rest-api-specs/pull/2271#issuecomment-360929670), this PR should ideally just be merged into the existing network 2018 PR (if it still exists)

olydis commented:
@nathantypanski note that *all* Swaggers you copied to the new API version folder should also have the corresponding API version - if I see correctly, you've only updated the one in `usage.json`

olydis commented:
@joannakleong @DeepakRajendranMsft @dedhar 
Please review this PR and add the changes of this PR (type fix) to your own changes. AFAIK you have a PR for 2018-??-?? open somewhere anyways - let's combine our efforts so there won't be any merge conflict 🙂

nathantypanski commented:
My bad! I’ll fix this ASAP. (I’m new to this process, so thanks for your
patience)

On Fri, Jan 26, 2018 at 6:00 PM Johannes Bader <notifications@github.com>
wrote:

> @nathantypanski <https://github.com/nathantypanski> note that *all*
> Swaggers you copied to the new API version folder should also have the
> corresponding API version - if I see correctly, you've only updated the one
> in usage.json
>
> —
> You are receiving this because you were mentioned.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/Azure/azure-rest-api-specs/pull/2271#issuecomment-360928536>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABUitD3kGYIqsBhTejWFfKJJJ97pZ4TVks5tOlkZgaJpZM4RgdBf>
> .
>
-- 

        Nathan Typanski

lmazuel reviewed (changes requested):
Invalid Python conf

DeepakRajendranMsft commented:
Please cancel this PR and create one against 2018-02-01

marstr commented:
Howdy @nathantypanski, I think you've caught us off-guard a little because we're not used to external contributors being so helpful! A lot of the advice and feedback in this thread has been given to you assuming you are a member of the Network team, and have control over server-side deployments. Seeing as you do not, we need to change tack here a little. This is a good bug, and we need to investigate it further. I'm going to close this PR and instead open an issue, which I'll assign to myself and drive. I'll need your help to fill in some of the details. Sorry for all of the confusion!

AutorestCI commented:
This commit was treated and no generation was made for Azure/azure-sdk-for-go

AutorestCI commented:
<details><summary>Swagger to SDK encountered an unknown error: (Azure/azure-sdk-for-go)</summary><p>

```python
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/dist-packages/swaggertosdk/github_tools.py", line 29, in exception_to_github
    yield context
  File "/usr/local/lib/python3.6/dist-packages/swaggertosdk/restapi/github.py", line 180, in rest_handle_action
    return rest_pull_close(body, github_con, restapi_repo, sdk_pr_target_repo, sdkbase)
  File "/usr/local/lib/python3.6/dist-packages/swaggertosdk/restapi/github.py", line 311, in rest_pull_close
    rest_pr.create_issue_comment("Was unable to create SDK %s PR for this closed PR.", sdkid)
TypeError: create_issue_comment() takes 2 positional arguments but 3 were given

```

</p></details>

nathantypanski commented:
Sure. I’ll gladly provide any details that could help. I’m currently
running a fork of the generated Go SDK to work around this bug, and it’d be
nice to see the fix land upstream.

On Tue, Jan 30, 2018 at 12:42 PM Martin Strobel <notifications@github.com>
wrote:

> Howdy @nathantypanski <https://github.com/nathantypanski>, I think you've
> caught us off-guard a little because we're not used to external
> contributors being so helpful! A lot of the advice and feedback in this
> thread has been given to you assuming you are a member of the Network team,
> and have control over server-side deployments. Seeing as you do not, we
> need to change tack here a little. This is a good bug, and we need to
> investigate it further. I'm going to close this PR and instead open an
> issue, which I'll assign to myself and drive. I'll need your help to fill
> in some of the details. Sorry for all of the confusion!
>
> —
> You are receiving this because you were mentioned.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/Azure/azure-rest-api-specs/pull/2271#issuecomment-361673612>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABUitMRMxPDiu-1bJ5mX0p7_WsjWhzdLks5tP1REgaJpZM4RgdBf>
> .
>
-- 

        Nathan Typanski

AutorestCI commented:
Did a commit to Azure/azure-sdk-for-python:
https://github.com/Azure/azure-sdk-for-python/commit/cf10fc70279c0073cf32afd3ae31e9356eefb250
